transfers/views.py

```diff
@@ -1,3 +1,36 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.contrib.auth.forms import AuthenticationForm
+from django.views.decorators.csrf import csrf_protect
+from django.contrib.auth import login, authenticate
+from django.contrib import messages
+from .models import UserProfile
+from .constants import UserType
 
-# Create your views here.
+
+@csrf_protect
+def login_view(request):
+    if request.method == 'POST':
+        form = AuthenticationForm(request.POST)
+        username = request.POST['username']
+        password = request.POST['password']
+        user = authenticate(request, username=username, password=password)
+        if user is not None:
+            login(request, user)
+            profile = UserProfile.objects.get(user=user)
+            if (profile.user_type == UserType.STUDENT.value):
+                print("not a student")
+                return redirect('student_dashboard')
+        else:
+            messages.error(request, 'username or password not correct')
+    else:
+        form = AuthenticationForm()
+    return render(request, 'transfers/login.html', {'form': form})
```

Comment:
Again, as said. No need to write all this again. Every django built-in app already has this functionality via `contrib.auth.views`. All we need is a redirect view, which will check the type of user and redirect it to its respective home-page.

So a user of type student will be redirected to `student_dashboard` view via this `redirect` view.